### PR TITLE
Add Camera implementation using CameraView on Android

### PR DESCRIPTION
**Pull request by sibelius**
This is most a copy of expo camera module

this still half baked, missing javascript part and package and modules
fixes

see #246

the idea is to call this as RNCamera, and keep the RCTCamera as the old one

### CODE REVIEW

File: android/src/main/java/org/reactnative/camera/CameraViewManager.java [hunk 1]
```diff
@@ -41,10 +41,10 @@ public String toString() {
     }
   }
 
-  private static final String REACT_CLASS = "ExponentCamera";
+  private static final String REACT_CLASS = "RNnentCamera";
```
Comment (kraffslol) [inline android/src/main/java/org/reactnative/camera/CameraViewManager.java]: RNnentCamera? 😄

**Comment by ptomasroos** [comment]
I'm on vacation will check on Monday!

**Comment by sibelius** [comment]
we can go in 2 directions on this PR:

- [ ] use the interface closer to expo camera api, 
- [ ] or make this closer to current react-native-camera interface

going closer to expo camera api will make easier to keep expo camera modules changes in this package, and also easy to port some improvements of this package to expo itself. We could create a compatible interface from javascript to make easy to upgrade to the new implementation

**Comment by ospfranco** [comment]
with the next release of rn (0.52) the createJSModules should no longer be supported, as we are going forward with this, we should also take care of it, #1039

**Comment by wassgha** [comment]
@sibelius just to help with the decision, some of the features I've noticed are missing from react-native-camera but are present in the expo camera api are:
- Manual zoom
- Manual focus
- Face detection
- Separate still photo / video recording endpoints
We'd totally love to see these features in react-native-camera.

**Comment by afilp** [comment]
@sibelius It is possible to first release a new npm version that handles the frequent crash problems on Android? I saw that you created a PR for this, can this be merged/released and then enhance the features? Thanks!

**Comment by sibelius** [comment]
@afilp can you at least run both examples to make sure everything is working fine on android?

**Comment by wassgha** [comment]
There's a `cameraview` dependency missing, I am getting
```
* What went wrong:
A problem occurred configuring project ':app'.
> Could not resolve all dependencies for configuration ':app:_debugApkCopy'.
   > Could not find com.github.react-native-community:cameraview:df60b07573.
     Required by:
         project :app > project :react-native-camera
```

**Comment by sibelius** [comment]
we need to improve README

we need jitpack to run properly

https://github.com/react-native-community/react-native-camera/blob/master/RNCameraExample/android/build.gradle#L27

**Comment by mckn** [comment]
What was the final decision regarding this PR?

**Comment by sibelius** [comment]
@mckn I've decided to add this new implementation as RNCamera, so this won't break anything for old users, that will keep to use RCTCamera

We also need to bring the RNCamera implementation to ios part based on this https://github.com/react-native-community/react-native-camera/issues/1084

let me know if you wanna help us doing this

We can chat more about it on expo slack (https://slack.expo.io/)

**Comment by DevarakondaV** [comment]
For some reason
`react-native run-android`

throws several errors of the same variation.

`C:\Users\devar\Documents\React\SnapMe\node_modules\react-native-camera\android\src\main\java\org\reactnative\facedetector\tasks\FileFaceDetectionAsyncTask.java:81: error: incompatible types: String cannot be converted to Throwable
      mPromise.reject(ERROR_TAG, "The file does not exist. Given path: `" + mPath + "`.");`

and 

`C:\Users\devar\Documents\React\SnapMe\node_modules\react-native-camera\android\src\main\java\org\reactnative\camera\events\BarCodeReadEvent.java:27: error: no suitable method found for init(int)
    event.init(viewTag);
         ^
    method Event.init(int,long) is not applicable
      (actual and formal argument lists differ in length)
    method BarCodeReadEvent.init(int,Result) is not applicable
      (actual and formal argument lists differ in length)`

I've looked at the implementation for the first error in Promise.java and Promise.reject is an overloaded function. For some reason it still seems to fail. I also tried passing the second argument as a Throwable. This leads to a different error. 

The second error also doesn't make much sense because the Even.init function is also implemented.

Any insights to what is causing this?

**Comment by sibelius** [comment]
we decided to go in this way

- [x] use the interface closer to expo camera api,

so it will be easy to move from and to expo

**Comment by wassgha** [comment]
Could you confirm if `recordAsync` works in this version? It did not work for me.

**Comment by sibelius** [comment]
open an issue, and we can track this

**Pull request merged**